EclEditor/LexGENERAL.cxx

```diff
@@ -0,0 +1,173 @@
+    WordList &hthorList = *keywordlists[2];
+    WordList &roxieList = *keywordlists[3];
+    WordList &localList = *keywordlists[4];
+
+    char s[1000];
+    bool checkword = false;
+    int linetype = 0;
+
+    StyleContext sc(startPos, length, initStyle, styler);
+    int line = 0;
+
+    for (; sc.More(); sc.Forward()) {
+        sc.GetCurrentLowered(s, sizeof(s));
+        strtrim(s);
+
+        // Determine if the current state should terminate.
+        switch (sc.state) {
+        case SCE_GEN_HEADER:
+            if (sc.chPrev == ':') {
+                sc.ChangeState(SCE_GEN_DEFAULT);
+                sc.SetState(SCE_GEN_DELIMITER);
+                if (linetype == SCE_GEN_CARET)
+                    sc.ForwardSetState(SCE_GEN_CARET);
+                else
+                    sc.ForwardSetState(SCE_GEN_DEFAULT);
+            }
+            break;
+        case SCE_GEN_CARET:
+        case SCE_GEN_THOR_BACKGROUND:
+        case SCE_GEN_HTHOR_BACKGROUND:
+        case SCE_GEN_ROXIE_BACKGROUND:
+        case SCE_GEN_LOCAL_BACKGROUND:
+            if (sc.atLineEnd)
+            {
+                sc.SetState(SCE_GEN_DEFAULT);
+            }
+            break;
+        }
+
+        if (sc.state == SCE_GEN_DEFAULT) {
+            checkword = false;
+            if (headerList.InList(s)) {
+                if (stricmp(s, "caret") == 0)
+                {
+                    sc.SetState(SCE_GEN_HEADER);
+                    linetype = SCE_GEN_CARET;
+                }
+                else if (stricmp(s, "target") == 0)
+                {
+                    sc.ChangeState(SCE_GEN_HEADER);
+                    linetype = SCE_GEN_TARGET;
+                }
+            }
+            else {
+                if (thorList.InList(s)) {
+                    sc.ChangeState(SCE_GEN_THOR_BACKGROUND);
+                    sc.SetState(SCE_GEN_THOR_BACKGROUND);
+                }
+                else if (hthorList.InList(s)) {
+                    sc.ChangeState(SCE_GEN_HTHOR_BACKGROUND);
+                    sc.SetState(SCE_GEN_HTHOR_BACKGROUND);
+                }
+                else if (roxieList.InList(s)) {
+                    sc.ChangeState(SCE_GEN_ROXIE_BACKGROUND);
+                    sc.SetState(SCE_GEN_ROXIE_BACKGROUND);
+                }
+                else if (localList.InList(s)) {
+                    sc.ChangeState(SCE_GEN_LOCAL_BACKGROUND);
+                    sc.SetState(SCE_GEN_LOCAL_BACKGROUND);
+                }
+            }
+        }
+    }
+    sc.Complete();
+}
+
+static void FoldGENERALDoc(unsigned int startPos, int length, int initStyle,
+    WordList *[], Accessor &styler) {
+}
+
+static const char * const generalWordLists[] = {
+            "Headers",
+            "Thor",
+            "HThor",
+            "Roxie",
+            "Local",
+            0,
+        };
+
+LexerModule lmGENERAL(SCLEX_GENERAL, ColouriseGENERALDocSensitive, "general", FoldGENERALDoc, generalWordLists);
```

Comment:
The general lexer is for non-language related color selections. I tried coding the "general" color settings without the lexer. After a 1 day trip down that road without success and everything broken beyond repair and only getting worse, I decided the easiest way to implement the non-language specific color preferences was to create a "general" lexer. It solved that problem quite nicely.